File: 4107-find-missing-elements/find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        nums.sort()
        result = []
        for i in range(len(nums) - 1):
            gap = nums[i + 1] - nums[i]
            if gap > 1:
                result.extend(range(nums[i] + 1, nums[i + 1]))
        return result

File: 4107-find-missing-elements/find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<int> result;
        for (int i = 0; i < nums.size() - 1; i++) {
            int gap = nums[i + 1] - nums[i];
            if (gap > 1) {
                for (int x = nums[i] + 1; x < nums[i + 1]; x++) {
                    result.push_back(x);
                }
            }
        }
        return result;
    }
};